### ISSUE TITLE
image or video bigger than 1.2M will failed with beloe error info

### ISSUE THREAD

**Issue by marvin520** [Clarifai/clarifai-python#36]
> ApiError:{u'status_code': u'ALL_ERROR', u'status_msg': u'All images in request have failed. Please review the error messages per image.', u'meta': {}, u'results': [{u'docid': 6969405136956554562, u'url': u'http://video-test.eengoo.com/JUFNERPILCQEHCZXSAZVCKZBYVUAQUXF.mp4/14/oss-cn-hangzhou.aliyuncs.com?OSSAccessKeyId=DmbQjIq5KjcRWJ6A&Expires=1477453930&Signature=F3lI%2FKaQsy23JdOHy9ZO1VpbSjU%3D', u'status_code': u'CLIENT_ERROR', u'status_msg': u'Data loading failed, see results for details. ', u'local_id': u'', u'result': {u'error': u"'url' could not be downloaded 'http://video-test.eengoo.com/JUFNERPILCQEHCZXSAZVCKZBYVUAQUXF.mp4/14/oss-cn-hangzhou.aliyuncs.com?OSSAccessKeyId=DmbQjIq5KjcRWJ6A&Expires=1477453930&Signature=F3lI%2FKaQsy23JdOHy9ZO1VpbSjU%3D'"}, u'docid_str': u'0b4007f252be3f5160b84d0bec13fd42'}]}

i also use curl to do same thing, and found that if i use curl post, it is ok:
`
curl "https://api.clarifai.com/v1/tag/" \
  -X POST --data-urlencode "url=http://yinnut-video-test.oss-cn-hangzhou.aliyuncs.com/14%2F2ec26f9bceb7fea5c771c735f8759c45%2FJUFNERPILCQEHCZXSAZVCKZBYVUAQUXF.mp4?OSSAccessKeyId=DmbQjIq5KjcRWJ6A&Expires=1477485065&Signature=FbolxWKyihbYtcu0caBtfmgMtyE%3D" \
  -H "Authorization: Eengoo {token}"
`
but while i use curl get, it will not work:
`
curl "https://api.clarifai.com/v1/tag/?url=http://yinnut-video-test.oss-cn-hangzhou.aliyuncs.com/14%2F2ec26f9bceb7fea5c771c735f8759c45%2FJUFNERPILCQEHCZXSAZVCKZBYVUAQUXF.mp4?OSSAccessKeyId=DmbQjIq5KjcRWJ6A&Expires=1477485065&Signature=FbolxWKyihbYtcu0caBtfmgMtyE%3D" \
  -H "Authorization: Eengoo {token}"
`

i also notice that, in your demo web, if i upload image or video from local, it will be ok, and if i use url, it will not work. i have send emails to your support email box, please help to fix this proble.

or if there are some selection that i can use tag_image_urls API with POST in your code, please inform me.
here is my code:
`
      result = self.clarifai_api.tag_image_urls(urls=video_url,language=language)
                if result['status_code']=='TOKEN_EXPIRED':
                    self.update_token()
                    result = self.clarifai_api.tag_image_urls(urls=video_url,language=language)
            assert result['status_code'] == 'OK'
            assert result['results'][0]['url'] == video_url
        except ApiError as e:
            self.logger.error('calrifai tags error of {0} with {1}'.format(video_url,e))
            self.logger.error(traceback.format_exc())
            return
`

**Comment by marvin520** [comment]
`---
Metadata-Version: 1.0
Name: clarifai
Version: 2.0.10
Summary: Clarifai API Python Client
Home-page: https://github.com/clarifai/clarifai-python
Author: Robert Wen
Author-email: robert@clarifai.com
License: Apache 2.0
Location: /Users/chaoxu/clarifai_env/lib/python2.7/site-packages/clarifai-2.0.10-py2.7.egg
Requires: future, requests, configparser
Classifiers:
`

is this version is right ?
 i use `pip install clarifai==2.0.10` to install as well as setup.py. when i use `from clarifai.rest import ClarifaiApp`, it report error:

>>> from clarifai.rest import ClarifaiAp
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named rest
>>>

**Comment by robertwenquan** [comment]
@marvin520 Thanks for reporting the issue. We are taking it very seriously.

For the investigation, I used the url "http://yinnut-video-test.oss-cn-hangzhou.aliyuncs.com/14/2ec26f9bceb7fea5c771c735f8759c45/JUFNERPILCQEHCZXSAZVCKZBYVUAQUXF.mp4" and managed to reproduce the problem. 

The test I had ended up with slightly different result as you reported. I could not tag the url with either curl -X POST or curl -X GET or the python client api.

The short answer is our server has a 60 sec timeout for the POST request and 30 second timeout for the GET request. I tested download with the above url, it takes about 55 seconds to finish with the wget. That's probably why you went through the POST but not the GET.

I tried to wget the above url. It ended up downloaded about 55 seconds. It's really close to the 60 sec threshold so I assume when sometimes the network fluctuate it will fail even with the POST request. 

![image](https://cloud.githubusercontent.com/assets/8893623/19917623/26ed4f28-a09b-11e6-95e2-fed706a91c96.png)

Here is my test result. Basically all the curl -X POST, curl -X GET and the python client failed.

Furthermore, I also tried to download the file and host it on our aws webserver. The same file could be tagged with the aws http url because the fetch time then becomes less than 1 second.

```
time curl -X POST 'https://api.clarifai.com/v1/tag/' --data-urlencode "url=http://yinnut-video-test.oss-cn-hangzhou.aliyuncs.com/14/2ec26f9bceb7fea5c771c735f8759c45/JUFNERPILCQEHCZXSAZVCKZBYVUAQUXF.mp4" -H "Authorization: Bearer [redacted]"
{
  "status": {
    "code": 10020,
    "description": "Bad gateway",
    "details": "Please try again or contact support@clarifai.com if the problem persists. code:2502"
  }
}

0.02s user
0.01s system
1:00.06 total
```

```
time curl 'https://api.clarifai.com/v1/tag/?url=http://yinnut-video-test.oss-cn-hangzhou.aliyuncs.com/14/2ec26f9bceb7fea5c771c735f8759c45/JUFNERPILCQEHCZXSAZVCKZBYVUAQUXF.mp4' -H "Authorization: Bearer [redacted]"

{"status_code": "ALL_ERROR", "status_msg": "All images in request have failed. Please review the error messages per image.", "meta": {}, "results": [{"docid": 16907448361921909200, "url": "http://yinnut-video-test.oss-cn-hangzhou.aliyuncs.com/14/2ec26f9bceb7fea5c771c735f8759c45/JUFNERPILCQEHCZXSAZVCKZBYVUAQUXF.mp4", "status_code": "CLIENT_ERROR", "status_msg": "Data loading failed, see results for details. ", "local_id": "", "result": {"error": "'url' could not be downloaded 'http://yinnut-video-test.oss-cn-hangzhou.aliyuncs.com/14/2ec26f9bceb7fea5c771c735f8759c45/JUFNERPILCQEHCZXSAZVCKZBYVUAQUXF.mp4'"}, "docid_str": "3f3517ffa832c22aeaa352b4a69d91d0"}]}

0.03s user
0.01s system
29.883 total
```

The following code snippet

```
from clarifai.client import ClarifaiApi

api = ClarifaiApi()

import time

print time.asctime()

try:
    api.tag_image_urls(urls='http://yinnut-video-test.oss-cn-hangzhou.aliyuncs.com/14/2ec26f9bceb7fea5c771c735f8759c45/JUFNERPILCQEHCZXSAZVCKZBYVUAQUXF.mp4')
except:
    pass

print time.asctime()
```

runs as 

```
$ python test.py
Wed Nov  2 01:20:09 2016
Wed Nov  2 01:21:10 2016
```

To conclude this problem, there is still debate whether this is a defect or not. It is suggested to use the POST method as it has higher timeout value. We can not up the timeout value to infinite because it will be prone to DoS attack and it's really hard to figure out how slow the customer's network will be. 20KB/s is pretty much very slow internet from server to server. In the near future we will proably host some api server in aws Asian regions to address this kind of fetching latency problem. But before that happens, it's suggested on your side to make your webserver as fast as possible for a fetch from US servers, either through CDN, or a proxy. 

Lastly, have you ever tried use the local file upload api call?

```
from clarifai.client import ClarifaiApi
api = ClarifaiApi()
api.tag([open('file.mp4')])
```

I know it perhaps will make no difference because it still uses the uplink bandwidth as the POST uses, but as I didn't see u used it probably you still want a try.

At the last resort, you will have to provide video snippet smaller to make sure the fetch never exceeds 60 sec. I will see if we can up the timeout value but this really needs more consideration and takes some time to happen. 

Thanks for your understanding.

Robert

**Comment by robertwenquan** [comment]
@marvin520 as for the second question. It is not related to the original problem. Are you still having that problem? Please open a new issue if it's still bothering you. From what I can see it may be a misconfiguration of PYTHONPATH or something else. Let's address it in another issue as it's not related.

**Comment by robertwenquan** [comment]
cause determined. Not a defect. Closing this issue now. shoot me email if you need further help on this.
